lib/core/error.js

```diff
@@ -102,9 +102,7 @@ class MongoNetworkError extends MongoError {
     super(message);
     this.name = 'MongoNetworkError';
 
-    if (options && options.beforeHandshake === true) {
-      this[kBeforeHandshake] = true;
-    }
+    this[kBeforeHandshake] = !!(options && options.beforeHandshake === true);
```

Comment:
Just to confirm, do we actually want to explicitly set the `kBeforeHandshake` symbol on all network errors regardless of whether the options/property is passed in?